### ISSUE TITLE
notifications?

### ISSUE THREAD

**Issue by elsehow** [elsehow/hyperchat2#3]
it would be nice to get notified when a new post comes in - one you've commented on.

however, it seems like this would require one of two things:
1. persistent identities (pseudonymous or no)
2. optional app state, stored in the level db. the app store can have a `subscribed-to` field, a list of message keys. when a new message comes in that's a reply-to to one of those keys, the user receives a pop-up notification.

**Comment by harperj** [comment]
You could also just have `subscribed-to` operate on identities, even if they aren't persistent.  This way you'll get notifications any time a certain pseudonym is used.  In that case, though, it seems that someone could easily spam your alerts.

**Comment by elsehow** [comment]
yes, i can assume an identity (pseudonym) and spam. also, identities are flexible, whereas hashes are guaranteed to persist, so its a little easier to do notifications for, at least for now

once there's a more robust identity system, i think that would be better. in that case, we could also have encrypted messaging (like facebook messages feature) and other good stuff.

**Comment by dmajoraddnine** [comment]
is it helpful to use Local Storage to hold the notifications data?  then identity doesn't matter and you just need to check post hashes as they come in.

**Comment by elsehow** [comment]
yes, we are holding everything in [level](https://www.npmjs.com/package/level-browserify). so, we can hold something in level like:

`var watching = [ postkey1, postkey2, ... ]`

then,

``` javascript
swarmlog.on('data', d => {
  if (store.watching.contains(d.value.key)
    notify(d)
}
```
